boards/adafruit-grand-central-m4-express/include/board.h

```diff
@@ -20,6 +20,7 @@
 #define BOARD_H
 
 #include "cpu.h"
+#include "periph/gpio.h"
```

Comment:
Could we add this to `led.h` instead to avoid changing all board definitions?

Reply:
Well we could, but if we want to follow the include-what-you-use paradigm (see #20570), it doesn't make sense to add it to `led.h`. That file does not use the content of `periph/gpio.h`, while the `board.h` files indeed do (e.g., `GPIO_PIN`).

Reply:
For example, running `make -C examples/hello-world BOARD=adafruit-grand-central-m4-express all` with the following diff also fails on `master` due to the missing include in `board.h`, completely independent of `led.h`:

```diff
diff --git a/examples/hello-world/main.c b/examples/hello-world/main.c
index 213128ac64..15992abe24 100644
--- a/examples/hello-world/main.c
+++ b/examples/hello-world/main.c
@@ -28,5 +28,7 @@ int main(void)
     printf("You are running RIOT on a(n) %s board.\n", RIOT_BOARD);
     printf("This board features a(n) %s CPU.\n", RIOT_CPU);
 
+    printf("board.h defined PIN: %d\n", WS281X_PARAM_PIN);
+
     return 0;
 }
```

This now only gets apparent because the macros are used in the inline functions.

But anyway, I feel like the `board.h` include structure could benefit from a bigger overhaul in any case, so I'm not too strictly opposed to put the include in `led.h` instead for now. What do you think?